src/main/kotlin/fi/metatavu/vp/deliveryinfo/thermometers/ThermometerController.kt

```diff
@@ -0,0 +1,123 @@
+package fi.metatavu.vp.deliveryinfo.thermometers
+
+import fi.metatavu.vp.deliveryinfo.devices.Device
+import fi.metatavu.vp.deliveryinfo.sites.Site
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.inject.Inject
+import java.time.OffsetDateTime
+import java.util.*
+
+/**
+ * Thermometer controller
+ */
+@ApplicationScoped
+class ThermometerController {
+    @Inject
+    lateinit var thermometerRepository: ThermometerRepository
+
+    /**
+     * Create new thermometer
+     *
+     * @param hardwareSensorId hardware sensor id
+     * @param device device
+     * @param site site
+     * @return created thermometer
+     */
+    suspend fun createNew(
+        hardwareSensorId: String,
+        device: Device,
+        site: Site
+    ): Thermometer {
+        return thermometerRepository.create(
+            id = UUID.randomUUID(),
+            hardwareSensorId = hardwareSensorId,
+            espMacAddress = device.deviceId,
+            site = site
+        )
+    }
+
+    /**
+     * Archive the old thermometer if it exists
+     *
+     * @param thermometer thermometer
+     * @param device device
+     * @param site site
+     * @return true if archiving took place
+     */
+    suspend fun archiveOldThermometer(thermometer: Thermometer?, device: Device, site: Site): Boolean {
+        if (thermometer != null && (thermometer.espMacAddress != device.deviceId || thermometer.site!!.id != site.id)) {
+            val archivedAt = OffsetDateTime.now()
+            thermometerRepository.update(thermometer = thermometer, archivedAt = archivedAt, name = thermometer.name, modifierId = null)
+            return true
+        }
+
+        return false
+
```

Comment:
a bit easier to read

```suggestion
        val thermometerIsUnchanged = thermometer?.espMacAddress == device.deviceId && thermometer.site!!.id == site.id

        if (thermometer == null || thermometerIsUnchanged) {
          return false
        }

        val archivedAt = OffsetDateTime.now()
        thermometerRepository.update(
            thermometer = thermometer,
            archivedAt = archivedAt,
            name = thermometer.name,
            modifierId = null
        )

        return true
```